app/models/map_overlay.rb

```diff
@@ -1,4 +1,11 @@
 class MapOverlay < ApplicationRecord
   acts_as_list
   has_and_belongs_to_many :localities
+  has_many :annotations
```

Comment:
Add `dependent: :destroy`

Reply:
Also probably ought to prevent deletion of map overlay if it's got annotations. Naive users might think, oh I've replaced the images on the map layer, so I'll add a new map layer and delete the old one, and poof, no more annotations, but lots of unhappy users.